exercises/list-ops/src/example.h

```diff
@@ -14,34 +14,34 @@ typedef struct {
 // constructs a new list
 list_t *new_list(size_t length, list_value_t values[]);
 
-// append entries to a list and return the new list
-list_t *append_list(list_t * list1, list_t * list2);
+// concat two lists into one new list
+list_t *concat_list(const list_t * list1, const list_t * list2);
 
 // filter list returning only values that satisfy the filter function
-list_t *filter_list(list_t * list, bool(*filter) (list_value_t value));
+list_t *filter_list(const list_t * list, bool (*filter)(list_value_t value));
 
 // returns the length of the list
-size_t length_list(list_t * list);
+size_t length_list(const list_t * list);
 
 // return a list of elements whose values equal the list value transformed by
 // the mapping function
-list_t *map_list(list_t * list, list_value_t(*map) (list_value_t value));
+list_t *map_list(const list_t * list, list_value_t(*map) (list_value_t value));
 
 // folds (reduces) the given list from the left with a function
-list_value_t foldl_list(list_t * list, list_value_t initial,
+list_value_t foldl_list(const list_t * list, list_value_t initial,
                         list_value_t(*foldl) (list_value_t value,
                                               list_value_t initial));
 
 // folds (reduces) the given list from the right with a function
-list_value_t foldr_list(list_t * list, list_value_t initial,
+list_value_t foldr_list(const list_t * list, list_value_t initial,
                         list_value_t(*foldr) (list_value_t value,
                                               list_value_t initial));
 
 // reverse the elements of the list
-list_t *reverse_list(list_t * list);
+list_t *reverse_list(const list_t * list);
 
 // destroy the entire list
-// list will be a dangling pointer after calling this method on it
+// list will be a dangling pointer after calling this function on it
```

Comment:
I don't agree with this one. There's a difference between functions and methods, it's not just a matter of all functions in C being called "function". Methods are a specific type of function that take the object it acts upon as the first (or hidden) argument.

Reply:
```suggestion
// list will be a dangling pointer after calling this method on it
```

Reply:
I complained about "method" in a comment on my solution of this exercise because "method" is not mentioned in the C standard (at least not in the sense of "function" or "function operating on some object"). There was a discussion about that on v3 (I think) where they agreed that they didn't want to introduce technical terms from other languages.

But that's not a hill I want to die on.

Reply:
Method isn't a term from another language, though. It's a general term referring to functions defined for or on objects which this definitely is. To me saying we want to avoid it is like saying this shouldn't be done in an object oriented way since C doesn't have built in support for object orientation.

Reply:
Does that also apply to the `foldl()` and `foldr()` functions/methods? Is every function that takes an argument a method becaus it acts upon it?  
If you insist on "method" that's OK with me. I'm just trying to understand the argument.

Reply:
In the case of the fold methods, I think the word "function" refers to the function passed as an argument to the method so the terminology is accurate.